Core/GameEngine/Source/Common/System/MiniDumper.cpp

```diff
@@ -0,0 +1,732 @@
+		for (int i = 0; i < m_dumpObjectsSubState; ++i)
+		{
+			allocator = allocator->getNextDmaInList();
+		}
+
+		MemoryPoolAllocatedRange rawBlockRange = {0};
+		int rawBlocksInDma = allocator->getRawBlockCount();
+		if (m_dmaRawBlockIndex < rawBlocksInDma)
+		{
+			// Dump this block
+			allocator->fillAllocationRangeForRawBlockN(m_dmaRawBlockIndex, rawBlockRange);
+			memoryBase = reinterpret_cast<ULONG64>(rawBlockRange.allocationAddr);
+			memorySize = rawBlockRange.allocationSize;
+			++m_dmaRawBlockIndex;
+		}
+
+		if (rawBlocksInDma == m_dmaRawBlockIndex)
+		{
+			// Advance to the next DMA
+			++m_dumpObjectsSubState;
+			m_dmaRawBlockIndex = 0;
+			if (allocator->getNextDmaInList() == NULL)
+			{
+				// Done iterating through all the DMAs
+				m_dumpObjectsSubState = 0;
+				++m_dumpObjectsState;
+			}
+		}
+		break;
+	}
+	default:
+		// Done, set "no more stuff" values
+		m_dumpObjectsState = 0;
+		m_dumpObjectsSubState = 0;
+		m_dmaRawBlockIndex = 0;
+		memoryBase = 0;
+		memorySize = 0;
+		moreToDo = FALSE;
+		break;
+	}
+
+	return moreToDo;
+}
+#endif
+
+// Comparator for sorting files by last modified time (newest first)
+bool MiniDumper::CompareByLastWriteTime(const FileInfo& a, const FileInfo& b)
+{
+	return ::CompareFileTime(&a.lastWriteTime, &b.lastWriteTime) > 0;
+}
+
+void MiniDumper::KeepNewestFiles(const std::string& directory, const DumpType dumpType, const Int keepCount)
+{
+	// directory already contains trailing backslash
+	std::string searchPath = directory + DumpFileNamePrefix + static_cast<Char>(dumpType) + "*";
+	WIN32_FIND_DATA findData;
+	HANDLE hFind = ::FindFirstFile(searchPath.c_str(), &findData);
+
+	if (hFind == INVALID_HANDLE_VALUE)
+	{
+		if (::GetLastError() != ERROR_FILE_NOT_FOUND)
+		{
+			DEBUG_LOG(("MiniDumper::KeepNewestFiles: Unable to find files in directory '%s': error=%u", searchPath.c_str(), ::GetLastError()));
+		}
+
+		return;
+	}
+
+	std::vector<FileInfo> files;
+	do
+	{
+		if (findData.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY)
+		{
+			continue;
+		}
+
+		// Store file info
+		FileInfo fileInfo;
+		fileInfo.name = directory + findData.cFileName;
+		fileInfo.lastWriteTime = findData.ftLastWriteTime;
+		files.push_back(fileInfo);
+
+	} while (::FindNextFile(hFind, &findData));
+
+	::FindClose(hFind);
+
+	// Sort files by last modified time in descending order
+	std::sort(files.begin(), files.end(), CompareByLastWriteTime);
+
+	// Delete files beyond the newest keepCount
+	for (size_t i = keepCount; i < files.size(); ++i)
+	{
+		if (::DeleteFile(files[i].name.c_str()))
+		{
+			DEBUG_LOG(("MiniDumper::KeepNewestFiles: Deleted old dump file '%s'.", files[i].name.c_str()));
+		}
+		else
+		{
+			DEBUG_LOG(("MiniDumper::KeepNewestFiles: Failed to delete file '%s': error=%u", files[i].name.c_str(), ::GetLastError()));
+		}
+	}
+}
+
+bool MiniDumper::EndsWithCaseInsensitive(const WideChar* str, const WideChar* suffix)
```

Comment:
String helper functions would be a better fit for stringex.h in WWLib.

Perhaps add new general purpose endswith functions like AsciiString has them?

AsciiString::startsWith
AsciiString::startsWithNoCase
AsciiString::endsWith
AsciiString::endsWithNoCase

Implement them standalone for both const char* and const wchar_t*

Then also use them in AsciiString and UnicodeString. I can help you with that if needed. Better make as a separate change (another one haha).